File: 프로그래머스/2/42578. 의상/의상.py
# 프로그래머스 42578 의상 LV.2
from functools import reduce
from collections import Counter

def solution(clothes):
    cnt = Counter([kind for name, kind in clothes])
    answer = reduce(lambda x, y: x*(y+1), cnt.values(), 1) - 1
    return answer

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>();
        
        for (String[] cloth: clothes) {
            map.put(cloth[1], map.getOrDefault(cloth[1], 1) + 1);
        }
        
        for (Integer v: map.values()) {
            answer *= v;
        }
        
        return answer - 1;
    }
}